src/NHibernate/Engine/ISessionImplementor.cs

```diff
@@ -187,6 +187,11 @@ public interface ISessionImplementor
 		void BeforeTransactionCompletion(ITransaction tx);
 
 		/// <summary>
+		/// 
+		/// </summary>
+		void FlushBeforeTransactionCompletion();
```

Comment:
Were you first willing to call it from outside of the session and the forgotten it? Or is there something causing you to want to have it in the interface although not used yet?